### ISSUE TITLE
Keypad of Manual Alarm Control Panel missing when code_template is used

### ISSUE THREAD

**Issue by Jpsy** [home-assistant/frontend#7385]
## Checklist

- [ X ] I have updated to the latest available Home Assistant version.
- [ X ] I have cleared the cache of my browser.
- [ X ] I have tried a different browser to see if it is related to my browser.

## The problem

The numeric keypad of the Manual Alarm Control Panel appears correctly when the config var `code` is used to define the PIN.
The keypad does **not** show when the config var `code_template` is used instead. (See screenshots below.)

The basic functionality of entering the code and switching states works in both cases. The problem is that with `code_template` the code must be entered through the system keyboard, which is not at all desirable on e.g. a wall mounted visualization tablet.

## Expected behavior

The keypad should show no matter if `code_template` or  `code` is used.

## Steps to reproduce

Use this Lovelace card for both of the following examples:

    # Lovelace card
    type: alarm-panel
    states:
      - arm_home
    entity: alarm_control_panel.tv_zugang

Keypad shows as expected with PIN in `code`:

    # configuration.yaml
    alarm_control_panel:
      - platform: manual
        name: TV-Zugang
        code: '111'

![image](https://user-images.githubusercontent.com/1806058/96365412-b78d4500-1140-11eb-8dde-248924bc1314.png)

Keypad does **NOT** show with PIN in `code_template`:

    # configuration.yaml
    alarm_control_panel:
      - platform: manual
        name: TV-Zugang
        code_template: >-
          {% if true %}
            111
          {% else %}
            222
          {% endif %}

![image](https://user-images.githubusercontent.com/1806058/96365418-c247da00-1140-11eb-944c-c3e912ccb95f.png)

## Environment

- Home Assistant release with the issue:
Home Assistant 0.116.4
HassOS 5.3
64 bit version on Raspberry Pi 4b
- Last working Home Assistant release (if known):
(unknown)
- Browser and browser version:
Latest Chrome, latest FireFox, Safari
- Operating system:
MacOs, Android, FireTablet

## State of relevant entities

```yaml

```

## Problem-relevant configuration

```yaml

```

## Javascript errors shown in your browser console/inspector

```txt

```

**Comment by Jpsy** [comment]
I found more details of the problem:

If the PIN code is defined through `code` then the attribute `code_format` of the resulting entity is set to `number`.
If the PIN code is defined through `code_template ` then the attribute `code_format` of the resulting entity is set to `text`.

I bet the numeric keypad is only shown when that attribute is `number` - which makes sense as the keypad is "numeric".

I guess the real problem is that there seems to be no way to use `code_template` and force the resulting PIN code to be of type `number`.

**Comment by Jpsy** [comment]
And another detail (sorry for the bits, pieces and pretzels):

According to the [HA develop docs' page for Alarm Control Panel](https://developers.home-assistant.io/docs/core/entity/alarm-control-panel/#code-formats) the allowed values for the attribute `code_format` are None, Number and Any - but not "text". So maybe there is also a more generic issue with the value "text" that is returned when `code_template` is used.

**Comment by spacegaier** [comment]
Looking at the code, `Any` will never be returned even if `code` is used in this platform. It is either `none`, `number` or `text` (which I assume is what `Any` refers in the dev docs). Also all other alarm platforms that I checked do the same since they use the same coding constants.

https://github.com/home-assistant/core/blob/941453dca9e45dd8ab554d2efbc661cd69b2990b/homeassistant/components/manual/alarm_control_panel.py#L281-L287

So in the end, not a frontend issue, but a backend gap. If the backend were to detect that the code returned by the `code_template` is numeric and properly provide that info, the frontend would react correctly to it.

**Comment by Jpsy** [comment]
@spacegaier you are right. 
Unfortunately I do not seem to have permission to transfer this issue to the home-assistant / core repo.

**Comment by spacegaier** [comment]
Yes, only selected project members do (I don't either). I will ping someone.
